salt/orch/prepare-product-migration.sls

```diff
@@ -0,0 +1,104 @@
+{#- Get a list of nodes seem to be down or unresponsive #}
+{#- This sends a "are you still there?" message to all #}
+{#- the nodes and wait for a response, so it takes some time. #}
+
+{%- set default_batch = salt['pillar.get']('default_batch', 5) %}
+{#- tx_update_migration_available needs to be last, as it is still used as a selector til reboot #}
+{%- set migration_grains = ['tx_update_migration_notes', 'tx_update_migration_newversion', 'migration_in_progress', 'tx_update_migration_available'] %}
+
+{%- set nodes_down = salt.saltutil.runner('manage.down') %}
+{%- if nodes_down|length >= 1 %}
+# {{ nodes_down|join(',') }} seem to be down: skipping
+  {%- do salt.caasp_log.debug('CaaS: nodes "%s" seem to be down: ignored', nodes_down|join(',')) %}
+  {%- set is_responsive_node_tgt = 'not L@' + nodes_down|join(',') %}
+{%- else %}
+# all nodes seem to be up
+  {%- do salt.caasp_log.debug('CaaS: all nodes seem to be up') %}
+  {#- we cannot leave this empty (it would produce many " and <empty>" targets) #}
+  {%- set is_responsive_node_tgt = '*' %}
+{%- endif %}
+
+{%- set is_migrateable = is_responsive_node_tgt + ' and G@tx_update_migration_available:true' %}
+
+{%- set is_migrateable_admin = 'G@tx_update_migration_available:true and G@roles:admin' %}
+
+{%- set nodes_up = salt.saltutil.runner('manage.up')|reject("equalto", "ca") %}
+
+{% for node in nodes_up %}
+  # Maybe using the cache here might lead to problems, if it is out of date.
+  {%- set migration_grain = salt['saltutil.runner']('cache.grains', tgt=node)[node].get('tx_update_migration_available', false) %}
+  {%- do salt.caasp_log.debug('Checking {0} for migration: {1}'.format(node, migration_grain)) %}
+  {% if not migration_grain %}
+    {% do salt.caasp_log.warn('Not all nodes are migratable: {0} is missing'.format(node)) %}
+  {% endif %}
+{% endfor %}
```

Comment:
I would remove this block. Doing all of this just for maybe a `salt.caasp_log.warn` is an overkill...

Reply:
dropped ;-)